app/routes/recipes-generator.jsx

```diff
@@ -44,54 +36,209 @@ export default function RecipiesGenerator() {
+                boxWidth={470}
+                leftRowInfo={["No has hecho recetas"]}
+                rightRowInfo={[""]}
+              />
+            ) : (
+              <DataCardHeart
+                boxWidth={470}
+                rows={lastConsumedRecipes.map((recipe) => ({
+                  recipeId: recipe.recipe_id,
+                  leftText: `${recipe.title || "Unknown Recipe"} - Calorías: ${
+                    recipe.calories || "N/A"
+                  }`,
+                  rightText: recipe.createdAt
+                    ? new Date(recipe.createdAt).toLocaleDateString()
+                    : "Invalid Date",
+                  isFavorite: favoriteRecipes.some(
+                    (fav) => fav.recipe_id === recipe.recipe_id
+                  ),
+                }))}
+                onToggleFavorite={(recipeId) => {
+                  toggleFavorite(recipeId);
+                }}
+              />
+            )}
           </div>
+
+          {/* Recetas favoritas */}
           <div className="flex flex-col items-center">
-            <div className="text-3xl font-bold mb-[15px]">Recetas favoritas</div>
-            <DataCard page='recipes-generator' boxWidth={470} leftRowInfo={["Almuerzo", "Desayuno", "Desayuno", "Cena"]} rightRowInfo={["Arroz con pollo al pesto", "Yogurt con semillas y frutas", "Smoothie rojo", "Ensalada de Quinoa"]} />
-            <div className="h-[20px]"></div>
-            <SearchBar/>
+            <div className="text-3xl font-bold mb-[15px] font-kantumruy">
+              Recetas favoritas
+            </div>
+            {loading ? (
+              <DataCardNoG
+              boxWidth={470}
+              leftRowInfo={["Generando Recetas"]}
+              rightRowInfo={[" "]}
+            />
+            ) : favoriteRecipes.length === 0 ? (
+              <DataCardNoG
+              boxWidth={470}
+              leftRowInfo={["No tienes recetas favoritas"]}
+              rightRowInfo={[" "]}
+            />
+            ) : (
+              <DataCardHeart
+                boxWidth={470}
+                rows={favoriteRecipes.map((recipe) => ({
+                  recipeId: recipe.recipe_id,
+                  leftText: `${recipe.title || "Unknown Recipe"} - Calorías: ${
+                    recipe.calories || "N/A"
+                  }`,
+                  rightText: recipe.createdAt
+                    ? new Date(recipe.createdAt).toLocaleDateString()
+                    : "Invalid Date",
+                  isFavorite: true,
+                }))}
+                onToggleFavorite={(recipeId) => {
+                  toggleFavorite(recipeId);
+                }}
+              />
+            )}
           </div>
         </div>
-
-        {/* Background images and Logo */}
-        <img src={ellipseBackground} alt="elipse" className="absolute top-[-35%] left-[-10%] z-[-1]"/>
+         <img src={ellipseBackground} alt="elipse" className="absolute top-[-35%] left-[-10%] z-[-1]"/>
```

Comment:
In this line, there might be an extra indentation which might break the HTML Syntax. 
Please review!

Reply:
Arreglado.